### ISSUE TITLE
call store procedure prevent further sql statement from executing

### ISSUE THREAD

**Issue by seayerktai** [ibmdb/node-ibm_db#237]
I am currently building a node server that utilize ibm_db node_module.
I found some weird issues when I try to do select query after call statement.
This is what I use right now.
 
var Pool = require('ibm_db').Pool,
var pool = new Poo();
 
pool.open(myconfig, function(err,db){
     
db.query('DECLARE GLOBAL TEMPORARY TABLE TABLEA (PARAM INT NOT NULL) IN USRTBS4 WITH REPLACE ON COMMIT PRESERVE ROWS NOT LOGGED;',[],function(err, rows){
            if(err){
              console.log(err);
            }
            db.query('CREATE UNIQUE INDEX SESSION.TABLEA_INDX ON SESSION.TABLEA (PARAM );', [], function(err, rows) {
              if(err){
                console.log(err);
              }
              db.query(' call SCHEMA.storeprocedure(0);', [], function(err, rows) {
                if(err){
                  console.log(err);
                }
                db.query('SELECT * FROM SESSION.TABLEA WHERE PARAM = 15813350;', [], function(err, rows) {
                  console.log('query result, ',rows);
                });
              });
            });
          });
})
 
 
Basically what this code do is
first, create temp table
second,  call storeprocedure store procedure to fill out the temp table,
third, try to querry the temp table.
 
The issue I am facing now, It looks like the final select query never execute, The operation just stop after finishing the call statement. I did some test on the ibm_db, and it turned out if I disable the "call statment check" in the ibm_db/lib/odbc.js and make it always consider the query as a non-call statement. the whole thing runs fine and I am able to get the result.

**Comment by bimalkjha** [comment]
@seayerktai When I tried to execute your shared code, it is throwing below errors. Please share a working test program to repro the issue. Thanks.
```
{ Error: [IBM][CLI Driver][DB2/LINUXX8664] SQL0440N  No authorized routine named "SCHEMA.STOREPROCEDURE" of type "PROCEDURE" having compatible arguments was found.  SQLSTATE=42884

  errors: [],
  error: '[node-ibm_db] SQL_ERROR',
  message: '[IBM][CLI Driver][DB2/LINUXX8664] SQL0440N  No authorized routine named "SCHEMA.STOREPROCEDURE" of type "PROCEDURE" having compatible arguments was found.  SQLSTATE=42884\n',
  state: '42884' }
/Users/bimaljha/nodework/master/ibm_db2/test/test-issue-237.js:21
          console.log('query,',ctes,rows);
                               ^
ReferenceError: ctes is not defined
```

**Comment by bimalkjha** [comment]
Also, try removing the space before `call` in `db.query('  call SCHEMA.storeprocedure(0);', [], function(err, rows) {` and then execute. Thanks.

**Comment by bimalkjha** [comment]
@seayerktai Are you using the ibm_db@2.0.0? In latest code on github, I don't see the call statement check in Database.prototype.query. Please share the line no that you think is causing the problem. Thanks.

**Comment by seayerktai** [comment]
it is in odbc.js line 542     if(sql.search(/^call /i))

I am using version 1.0.2.   The reason I put a space before call is to trick the the above if statement to evaluate to false so I can get passed the call statement.
I originally did not add the space before call statement, and that is when I see the sql stop executing after the call statmenet.

I cannot share my code here because I am using ibm_db for a business product. for testing, you should be able to reproduce the issue by just changing the naming in my code to the actual table/schema in your environment

**Comment by bimalkjha** [comment]
@seayerktai In your shared repro script, there is no create procedure statement for SCHEMA.storeprocedure, so error is expected. Please share the definition for SCHEMA.storeprocedure. By just changing the name, test script is not working. Thanks.

**Comment by bimalkjha** [comment]
@seayerktai Please check using ibm_db version 2.0.0. This issue might be fixed there as many such issues are fixed in v2.0.0. Thanks.
